### PR TITLE
Add php 7.2 linter and fix errors

### LINKED ISSUE #3753: Unable to open calendar app in latest v3.0.0 nor run its cron job
Just upgraded to your latest version 3.0.0 and seeing this error:

```
ArgumentCountError: Too few arguments to function OCA\Calendar\BackgroundJob\CleanUpOutdatedBookingsJob::__construct(), 0 passed in /usr/local/lib/owncloud/lib/private/BackgroundJob/JobList.php on line 271 and exactly 3 expected in /usr/local/lib/owncloud/apps/calendar/lib/BackgroundJob/CleanUpOutdatedBookingsJob.php:42
Stack trace:
#0 /usr/local/lib/owncloud/lib/private/BackgroundJob/JobList.php(271): OCA\Calendar\BackgroundJob\CleanUpOutdatedBookingsJob->__construct()
#1 /usr/local/lib/owncloud/lib/private/BackgroundJob/JobList.php(215): OC\BackgroundJob\JobList->buildJob(Array)
#2 /usr/local/lib/owncloud/cron.php(121): OC\BackgroundJob\JobList->getNext()
#3 {main}
```

And a similar related one:

```
{"reqId":"K0WOYtgDjdJQLID6zM9P","level":3,"time":"2021-11-30T09:28:04+13:00","remoteAddr":"121.74.106.133","user":"michael.heuberger@infernaldarkness.com","app":"index","method":"GET","url":"/cloud/index.php/apps/calendar/","message":{"Exception":"ParseError","Message":"syntax error, unexpected ')'","Code":0,"Trace":[{"file":"/usr/local/lib/owncloud/lib/composer/composer/ClassLoader.php","line":322,"function":"Composer\\Autoload\\includeFile","args":["/usr/local/lib/owncloud/apps/calendar/lib//Controller/ViewController.php"]},{"function":"loadClass","class":"Composer\\Autoload\\ClassLoader","type":"->","args":["OCA\\Calendar\\Controller\\ViewController"]},{"function":"spl_autoload_call","args":["OCA\\Calendar\\Controller\\ViewController"]},{"file":"/usr/local/lib/owncloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":107,"function":"__construct","class":"ReflectionClass","type":"->","args":["OCA\\Calendar\\Controller\\ViewController"]},{"file":"/usr/local/lib/owncloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":126,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Calendar\\Controller\\ViewController"]},{"file":"/usr/local/lib/owncloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":455,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Calendar\\Controller\\ViewController"]},{"file":"/usr/local/lib/owncloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":427,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Calendar\\Controller\\ViewController"]},{"file":"/usr/local/lib/owncloud/lib/private/AppFramework/App.php","line":139,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Calendar\\Controller\\ViewController"]},{"file":"/usr/local/lib/owncloud/lib/private/Route/Router.php","line":309,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Calendar\\Controller\\ViewController","index",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"calendar.view.index"}]},{"file":"/usr/local/lib/owncloud/lib/base.php","line":1008,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/calendar/"]},{"file":"/usr/local/lib/owncloud/index.php","line":37,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/usr/local/lib/owncloud/apps/calendar/lib/Controller/ViewController.php","Line":118,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.55 Safari/537.36","version":"20.0.14.2","id":"61a5393a1cbb9"}
```

Using Nextcloud version 20.0.14 here.

Hope you guys know what it is. Cannot use my busy calendar now.

### PR DESCRIPTION
**Pull request by miaulalala**
Fixes #3753

### CODE REVIEW

File: tests/php/unit/Controller/ViewControllerTest.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ protected function setUp(): void {
 			$this->appointmentContfigService,
 			$this->initialStateService,
 			$this->appManager,
-			$this->userId,
```
Comment (ChristophWurst) [inline tests/php/unit/Controller/ViewControllerTest.php:80]: exclude the linter for the tests directory and leave test files untouched

Comment (miaulalala) [reply tests/php/unit/Controller/ViewControllerTest.php:80]: damn ok

File: composer.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "scripts": {
         "cs:fix": "php-cs-fixer fix",
         "cs:check": "php-cs-fixer fix --dry-run --diff",
-        "lint": "find . -name \\*.php -not -path './vendor/*' -print0 | xargs -0 -n1 php -l",
+        "lint": "find . -name \\*.php -not -path './vendor/*' -not -path './tests/*' -print0 | xargs -0 -n1 php -l",
```
Comment (raimund-schluessler) [inline composer.json:18]: @ChristophWurst Is it really a good idea to exclude the tests from linting completely? They don't get linted at all now, not only not linted with PHP 7.2.

Comment (tcitworld) [reply composer.json:18]: I guess they will break anyway when executing them, so it doesn't matter much.

Comment (raimund-schluessler) [reply composer.json:18]: Well, one could fix them for 7.2 as well 😉

Comment (ChristophWurst) [reply composer.json:18]: php7.2 is EOL. we will make this work for 3.0.x but it will be dropped for 3.1

Comment (raimund-schluessler) [reply composer.json:18]: > we will make this work for 3.0.x but it will be dropped for 3.1

Then maybe only merge this into the stable3.0 branch?

Comment (ChristophWurst) [reply composer.json:18]: we'll merge, backport and then drop nextcloud 20 and php7.2

**Comment by miaulalala** [comment]
@ChristophWurst what to do about the JSON_THROWS_ON_ERROR? It's a 7.3+ feature. We wrap it in a runtime exception and rethrow without catching, so IMO it would be "fine" to just let it fail.

**Comment by ChristophWurst** [comment]
> @ChristophWurst what to do about the JSON_THROWS_ON_ERROR? It's a 7.3+ feature. We wrap it in a runtime exception and rethrow without catching, so IMO it would be "fine" to just let it fail.

Drop the php7.3 code as you already did :+1:

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**